js/libs/keycloak-admin-client/src/utils/fetchWithError.ts

```diff
@@ -1,8 +1,19 @@
-export type NetworkErrorOptions = { response: Response; responseData: unknown };
+type ResponseData =
+  | string
+  | number
+  | boolean
+  | null
+  | ResponseData[]
+  | { [key: string]: ResponseData };
+
+export type NetworkErrorOptions = {
+  response: Response;
+  responseData: ResponseData;
+};
 
 export class NetworkError extends Error {
   response: Response;
-  responseData: unknown;
+  responseData: ResponseData;
```

Comment:
I'd prefer we keep this `unknown` to be very honest. The type that replaces this is essentially what JSON could return, but either solution still requires an equivalent amount of defensive programming.

Reply:
Maybe you are right. I do not think it is the most critical thing compared to others.

The reasons why I proposed it is because previous version of keycloak had very different error handling.
And basically you do not know if you can use `JSON.stringify(responseData)`, who knows if responseData has circular dependencies, so for my case I needed to read source code to understand how can I use `responseData`. Does it contain methods or functions inside? Is it safe to serialize?

For me personally it was quite crucial because that was pain point for quite a while when working on error handling and logging using this library. Now I just use assertion `... as JsonValue` imported from `type-fest` the same as in this PR basically and adds the clarity of what type the responseData could be and it solves my problem. Also I do not know how others use it, so feel free to close this PR

Reply:
Yeah, you do have a point that this could be narrowed down. But I think the amount of value gained from it is relatively little and adds complexity to this code. I think for that reason I'd rather not land this at the moment, perhaps in the future this will change.

Thanks for your PR nonetheless!